tap_mysql/tap.py

```diff
@@ -42,10 +74,47 @@ class TapMySQL(SQLTap):
             secret=True,
             description="The password for the user",
         ),
+        th.Property(
+            "custom_streams",
+            th.ArrayType(custom_stream_config),
+            required=False,
+            description="An array of customized streams to use.",
+        ),
     ).to_dict()
 
+    def discover_streams(self) -> list[Stream]:
+        """Initialize all available streams and return them as a list.
+
+        Returns:
+            List of discovered Stream objects.
+        """
+        result: list[Stream] = []
+        custom_configs = self.config.get("custom_streams")
+        custom_stream_names = []
+        if custom_configs:
+            for stream in custom_configs:
+                for db_schema in stream.get("db_schemas"):
+                    custom_stream_names.append(f"{db_schema}-{stream['name']}")
+
+        for catalog_entry in self.catalog_dict["streams"]:
+            stream_id = catalog_entry["tap_stream_id"]
+            # if it's a custom stream treat it differently
+            if stream_id in custom_stream_names:
+                for stream in custom_configs:
+                    for db_schema in stream.get("db_schemas"):
+                        if stream_id == f"{db_schema}-{stream['name']}":
+                            result.append(
+                                self.custom_stream_class(
+                                    self, catalog_entry, stream, db_schema
+                                )
+                            )
+            else:
+                result.append(self.default_stream_class(self, catalog_entry))
+
+        return result
+
     # not supposed to do this but the logs of deselected streams are a drag
-    @final
+    @final  # type: ignore
```

Comment:
What does this do?

Reply:
I get a typing error in my linters because, technically, in Python you are not supposed to overwrite methods that are marked `@final`. That's kind of the whole point of marking it as `final`. Hence the note just above that line. You take a BIG performance hit when you accept the original method. Like it'll take 10 minutes+ in the case of our instances where we have thousands of tables to parse.